esrally/client.py

```diff
@@ -43,7 +42,30 @@ def __init__(self, hosts, client_options):
                 # advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings"
                 urllib3.disable_warnings()
             else:
+                # verify server certificates; order of properties set here matters!
+                self.ssl_context.verify_mode=ssl.CERT_REQUIRED
+                self.ssl_context.check_hostname = True
                 self.logger.info("SSL certificate verification: on")
+
+            # When using SSL_context, all SSL related kwargs in client options get ignored
+            client_cert = self.client_options.pop("client_cert", False)
+            client_key = self.client_options.pop("client_key", False)
+
+            if bool(client_cert) != bool(client_key):
+                self.logger.error(
+                    "Supplied client-options contain only one of client_cert/client_key. "
+                    "If your Elasticsearch setup requires client certificate verification both need to be supplied. "
+                    "See https://esrally.readthedocs.io/en/stable/command_line_reference.html?highlight=client_options#id2"
+                )
+                defined_client_ssl_option = "client_key" if client_key else "client_cert"
+                missing_client_ssl_option = "client_cert" if client_key else "client_key"
+                raise exceptions.InvalidSyntax("'{}' is missing from client-options but '{}' "
+                                               "has been specified".format(defined_client_ssl_option,
+                                                                           missing_client_ssl_option))
+            elif client_cert and client_key:
+                self.logger.info("SSL client authentication: on")
```

Comment:
Can we please also add a log statement when this feature is turned off (we do that for the other cases as well)?

Reply:
We can, but I think it would be a bit weird to warn about something when no corresponding arguments have been passed. I mean this would appear when `client_cert` + `client_key` haven't been passed, so, maybe it would be too much info for the user to see that a certain feature hasn't been enabled when the arguments influencing it haven't been passed to begin with?